### ISSUE TITLE
ListView renders a off pixel between a list item and a separator on Android

### ISSUE THREAD

**Issue by njovy** [flutter/flutter#44008]
## Steps to Reproduce
```dart

import 'package:flutter/material.dart';
void main() => runApp(MyApp());
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        backgroundColor: Colors.red,
        body: ListView.separated(
            itemBuilder: (context, _) => Container(
              color: Colors.white,
              child: ListTile(
                title: Text("TEST"),
              ),
            ),
            separatorBuilder:  (context, _) => Container(
              height: 1,
              color: Colors.white,
              padding: EdgeInsetsDirectional.only(start: 72),
              child: Container(height: 1, color: Colors.grey[100],width: double.infinity,),
            ),
            itemCount: 50),
      ),
    );
  }
}
```
Above is a problematic code that renders a background pixel between each item and a separator. 

![Android Screen](https://user-images.githubusercontent.com/1792070/68048751-96abc400-fd24-11e9-98a2-cca44fcfd788.png)
Android Screen

![iOS Screen](https://user-images.githubusercontent.com/1792070/68048826-b642ec80-fd24-11e9-8f27-37a3a492b12b.jpeg)
iOS Screen

A background shouldn't be displayed as there is no padding or margin set among items. This only happens on Android. At first, I found this issue with SliverList but it happens with ListView as well. I tested this with a release mode but still renders a background color.

* Target Platform: Android
* Target OS version/browser: Api 28
* Devices: Samsung Galaxy 9+

## Logs

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.5, on Mac OS X 10.14.4 18E226, locale en-KR)
    • Flutter version 1.9.1+hotfix.5 at /Users/captainpark/Library/Flutter/sdk
    • Framework revision 1aedbb1835 (2 weeks ago), 2019-10-17 08:37:27 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/captainpark/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.1

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 34.0.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] IntelliJ IDEA Ultimate Edition (version 2019.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 40.2.4
    • Dart plugin version 192.7402

[✓] Connected device (1 available)
    • SM G965N • 21bd4fb43b037ece • android-arm64 • Android 9 (API 28)

• No issues found!

```

**Comment by HansMuller** [comment]
This looks like a known problem with antialiasing and mapping from logical pixels to device pixels. There's an open issue for a documentation fix: #14288

More info about the problem can be found here:
https://github.com/flutter/flutter/issues/14288#issuecomment-361375791
https://github.com/flutter/flutter/issues/17084#issuecomment-385718108
https://github.com/flutter/flutter/issues/15035#issuecomment-370028740

**Comment by njovy** [comment]
@HansMuller Ahh I will track this issue there. Thank you!
